tests/hla/test_run_svmpoller.py

```diff
@@ -0,0 +1,141 @@
+
+        Characteristics of these tests:
+
+        Success Criteria:
+            The SVM processing returns a value of 0: Success or 1: Failure
+
+        The input master_list file is a list of poller filenames, one filename per line.
+        Each poller file must be obtained from a specified directory and read to obtain the
+        names of the data files which need to be processed.
+
+        This test file can be executed in the following manner:
+            $ pytest -n # -s --basetemp=/internal/hladata/yourUniqueDirectoryHere --bigdata --slow
+              --master_list /internal/hladata/input/master_poller_list.txt test_run_svmpoller.py >&
+              test_svmpoller_output.txt &
+            $ tail -f test_svmpoller_output.txt
+          * The `-n #` option can be used to run tests in parallel if `pytest-xdist` has
+            been installed where `#` is the number of cpus to use. THIS IS NOT ADVISED FOR USE.
+          * Note: When running this test, the `--basetemp` directory should be set to a unique
+            existing directory to avoid deleting previous test output.
+          * A default master list exists in the tests/hla directory and contains 121 datasets.  This
+            is probably NOT the list you want to use, but it allows you to see what this file should
+            contain.
+
+    """
+    print("TEST_RUN_SVMPOLLER. Dataset: ", dataset)
+    output_name = dataset + '.ecsv'
+
+    current_dt = datetime.datetime.now()
+    print(str(current_dt))
+
+    subdir = ""
+    prevdir = os.getcwd()
+
+    # create working directory specified for the test
+    if not tmpdir.ensure(subdir, dir=True):
+        cpyurdir = tmpdir.mkdir(subdir).strpath
+    else:
+        curdir = tmpdir.join(subdir).strpath
+    os.chdir(curdir)
+
+    return_value = 1
+
+    try:
+
+        # The dataset variable is a fully qualified poller file (CSV) with no header
+        # Copy the poller file to the current working directory
+        shutil.copy2(dataset, ".")
+
+        # Get the poller file path, as well as its simple name
+        poller_path = "/".join(dataset.split("/")[:-1])
+        poller_file = dataset.split("/")[-1]
+        table = Table.read(poller_file, format="ascii")
+
+        # Column "col8" contains fully qualified constituent filenames for the single visit
+        # The fully qualified path designated in the poller file is the shared cache
+        file_list = table["col8"].tolist()
+
+        # Check if the files to be processed are in the same directory as poller
+        # directory, otherwise they need to be copied from the on-line cache
+        for full_filename in file_list:
+            filename = full_filename.split("/")[-1]
+            log.info("Looking for file {}".format(filename))
+            if os.path.exists(poller_path + "/" + filename):
+                shutil.copy2(poller_path + "/" + filename, ".")
```

Comment:
Generating a new filename would be a little more transparent with the use of 'os.path.join(poller_path, filename)'.

Reply:
/Users/mdelapena/repos/drizzlepac/drizzlepac